File: app/src/main/java/com/tobikster/medicreminder/data/DateTimeConverter.java
package com.tobikster.medicreminder.data;

import android.arch.persistence.room.TypeConverter;

import org.joda.time.LocalTime;

@SuppressWarnings("WeakerAccess")
class DateTimeConverter {
	private DateTimeConverter() {}

	@TypeConverter
	public static LocalTime fromMillis(final Long millis) {
		return LocalTime.fromMillisOfDay(millis);
	}

	@TypeConverter
	public static Long fromLocalTime(final LocalTime localTime) {
		return (long) localTime.getMillisOfDay();
	}
}

File: app/src/main/java/com/tobikster/medicreminder/data/reminders/DateTimeConverter.kt
package com.tobikster.medicreminder.data.reminders

import android.arch.persistence.room.TypeConverter
import org.threeten.bp.LocalTime

class DateTimeConverter {
	companion object {
		@JvmStatic
		@TypeConverter
		fun secondToLocalTime(second: Long): LocalTime = LocalTime.ofSecondOfDay(second)

		@JvmStatic
		@TypeConverter
		fun localTimeToSecond(time: LocalTime): Long = time.toSecondOfDay().toLong()
	}
}
